scene/gui/popup.cpp

```diff
@@ -126,6 +126,27 @@ Rect2i Popup::_popup_adjust_rect() const {
 		current.position.y = parent.position.y;
 	}
 
+	if (current.size.y > parent.size.y) {
+		current.size.y = parent.size.y;
+	}
+
+	if (current.size.x > parent.size.x) {
+		current.size.x = parent.size.x;
+	}
+
+	// Early out if max size not set.
+	if (get_max_size() <= Size2()) {
```

Comment:
You could call get_max_size() only once and keep the value locally here. Since it's doing a platform specific call, that would ensure it never does anything expensive several times.

Reply:
Yeah good point, calling so many `get` methods in the same scope is quite the code smell.